### ISSUE TITLE
ScrollablePositionedList builder itemsCount zero

### ISSUE THREAD

**Issue by vasilich6107** [google/flutter.widgets#78]
Hi. It's so disappointing. When `itemCount` is zero it executes builder with index `0`
The same goes when `itemCount: 1` 

```
ScrollablePositionedList.builder(
  itemCount: 0,
  itemBuilder: (context, index) =>
      Text('Item $index'),
)
```

Please fix this. Thanks

**Comment by iKrupp** [comment]
This may be a problem related to https://github.com/google/flutter.widgets/issues/77

I suppose the initial value for the scroll = 0. Therefore, he is trying to build this widget for the 0 index.

It seems that this widget lacks checking the current scrolling index with the number of items in the list

**Comment by jamesderlin** [comment]
This should be fixed by #79. Thanks!
